src/libtriton/ast/astContext.cpp

```diff
@@ -1043,5 +1052,58 @@ namespace triton {
       return this->astRepresentation.print(stream, node);
     }
 
+    SharedAbstractNode AstContext::simplify_concat(std::vector<SharedAbstractNode> exprs) {
```

Comment:
Can you add some comments at each steps to help me (and others) to understand the algorithm. What you are looking for, what/how you do it. etc. If you can also avoid to name variables like : `r`, `e`, `n`. If they can have a better name to quickly identify their role, it will be nice :)

Reply:
Done